### PR TITLE
DOCS-2282 Rogue API rewritten

### PR DESCRIPTION
**Pull request by MaximBashurov**

### CODE REVIEW

File: docs/latest/api-discovery/rogue-api.md [hunk 1]
```diff
@@ -1,34 +1,43 @@
-# Shadow, Orphan, and Zombie API <a href="../../about-wallarm/subscription-plans/#subscription-plans"><img src="../../images/api-security-tag.svg" style="border: none;"></a>
+# Shadow, Orphan, Zombie API <a href="../../about-wallarm/subscription-plans/#subscription-plans"><img src="../../images/api-security-tag.svg" style="border: none;"></a>
 
-[API Discovery](overview.md) allows uncovering rogue (shadow, orphan and zombie) APIs.
+The [API Discovery](overview.md) module automatically uncovers rogue (shadow, orphan and zombie) APIs by comparing the discovered API inventory with customers' provided specifications. You upload your API specifications in the [**API Specifications**](#upload-specifications-and-set-comparison-parameters) section and the module automatically highlights shadow, orphan and zombie APIs.
```
Comment (AnastasiaTWW) [inline docs/latest/api-discovery/rogue-api.md:3]: section and the module automatically highlights shadow, orphan and zombie APIs by comparing the specification to the live traffic.
+ try to make shorter

File: docs/latest/api-discovery/rogue-api.md [hunk 2]
```diff
@@ -1,34 +1,43 @@
...
 
-## Terminology
```
Comment (AnastasiaTWW) [inline docs/latest/api-discovery/rogue-api.md:5]: info blocks look massively. need to use some other approach to explain these entities

File: docs/latest/api-discovery/rogue-api.md [hunk 3]
```diff
@@ -1,34 +1,43 @@
...
+
+* Your actual API inventory (changes will be revealed by API Discovery)
+* Your own specification (new versions can arrive)
 
-To upload specifications and set rogue API search parameters:
+So to set up a constant monitoring of rogue API, you have an option of comparison **on the hourly basis**. To use the option, specification must be uploaded from URL. The specification itself will be updated before each comparison.
+
+To set up monitoring rogue APIs on hourly basis:
 
 1. Navigate to the **API Specifications** section and click **Upload specification**.
```
Comment (AnastasiaTWW) [inline docs/latest/api-discovery/rogue-api.md:42]: let's add the links to the section

File: docs/latest/api-discovery/rogue-api.md [hunk 4]
```diff
@@ -38,31 +47,46 @@ To upload specifications and set rogue API search parameters:
 
         You can change comparison settings at any moment later - after this the comparison will be re-done providing new results.
```
Comment (AnastasiaTWW) [inline docs/latest/api-discovery/rogue-api.md:48]: re-done should not be divided into 2 lines

File: docs/latest/api-discovery/rogue-api.md [hunk 5]
```diff
@@ -38,31 +47,46 @@ To upload specifications and set rogue API search parameters:
...
+## Find rogue APIs by one-time comparison
 
-* Use the **Rogue APIs** filter to see only shadow, orphan and/or zombie APIs related to the selected comparisons and filter out the remaining endpoints.
+You can upload your specification to perform immediate one-time comparison of its content with endpoints revealed by API Discovery up to the moment. To do so, in the comparison settings use upload from your local machine variant or deselect the **Perform regular comparison** option for specification uploaded from URL.
 
-In the details of such endpoints, in the **Specification conflicts** section, the specification(s) with the help of which shadow/zombie/orphan was detected will be indicated.
+Consider the following:
 
-Shadow APIs are also displayed among the riskiest endpoints at the [API Discovery Dashboard](dashboard.md).
+* You can re-start comparison at any moment manually via specification menu → **Restart comparison**.
```
Comment (AnastasiaTWW) [inline docs/latest/api-discovery/rogue-api.md:75]: 1. delete "manually"
2. check if there are other places in the article where the short UI path is divided into several lines - try to get rid of it

File: docs/latest/api-discovery/rogue-api.md [hunk 6]
```diff
@@ -38,31 +47,46 @@ To upload specifications and set rogue API search parameters:
...
+In the details of such endpoints, in the **Specification conflicts** section, the specification(s) with the help of which shadow/zombie/orphan was detected will be indicated.
 
-* Use the **Compare to...** filter to select specification comparisons - only for them the rogue APIs will be highlighted by the special marks in the **Issues** column.
+Shadow APIs are also displayed among the riskiest endpoints at the [API Discovery Dashboard](dashboard.md).
 
-    ![API Discovery - highlighting and filtering rogue API](../images/about-wallarm-waf/api-discovery/api-discovery-highlight-rogue.png)
+## Find rogue APIs by one-time comparison
 
-* Use the **Rogue APIs** filter to see only shadow, orphan and/or zombie APIs related to the selected comparisons and filter out the remaining endpoints.
+You can upload your specification to perform immediate one-time comparison of its content with endpoints revealed by API Discovery up to the moment. To do so, in the comparison settings use upload from your local machine variant or deselect the **Perform regular comparison** option for specification uploaded from URL.
```
Comment (AnastasiaTWW) [inline docs/latest/api-discovery/rogue-api.md:71]: try to make shorter. you can use chatgpt fpr help. and also "in the comparison settings use upload from your local machine variant" is not really grammatically correct. it shpuld be at least "in the comparison settings, use the **upload from your local machine** variant"

File: docs/latest/api-discovery/rogue-api.md [hunk 7]
```diff
@@ -38,31 +47,46 @@ To upload specifications and set rogue API search parameters:
...
 
-In the details of such endpoints, in the **Specification conflicts** section, the specification(s) with the help of which shadow/zombie/orphan was detected will be indicated.
+Consider the following:
 
-Shadow APIs are also displayed among the riskiest endpoints at the [API Discovery Dashboard](dashboard.md).
+* You can re-start comparison at any moment manually via specification menu → **Restart comparison**.
+* Specifications uploaded from a local machine and having the same name are considered to be different versions of the same specification.
```
Comment (AnastasiaTWW) [inline docs/latest/api-discovery/rogue-api.md:76]: should not it be described in the block below?

Comment (MaximBashurov) [reply docs/latest/api-discovery/rogue-api.md:76]: It is usefull both here and there and presented in both places.

File: docs/latest/api-discovery/rogue-api.md [hunk 8]
```diff
@@ -1,68 +1,107 @@
...
+* You can download the previously uploaded specification via **API Specifications** → specification details window → **Download specification**.
 
-    ![API Discovery - API Specifications - uploading API specification to find rogue APIs](../images/about-wallarm-waf/api-discovery/api-discovery-specification-upload.png)
+## Specification versions and zombie APIs
 
-    Note that you can re-start comparison at any moment manually via specification menu → **Restart comparison**.
+Unlike shadow and orphan APIs, zombie APIs require comparison of different specification versions:
 
-1. Start uploading.
+* In case of [hourly automatic comparison](#monitor-rogue-apis-on-hourly-basis), just put new version to your URL - it will be processed by an hourly schedule or immediately if you select **Restart comparison** form the specification menu.
```
Comment (AnastasiaTWW) [inline docs/latest/api-discovery/rogue-api.md:66]: what URL?

Comment (AnastasiaTWW) [reply docs/latest/api-discovery/rogue-api.md:66]: URL where I host my spec?

File: docs/latest/api-discovery/rogue-api.md [hunk 9]
```diff
@@ -1,30 +1,12 @@
 # Shadow, Orphan, Zombie API <a href="../../about-wallarm/subscription-plans/#subscription-plans"><img src="../../images/api-security-tag.svg" style="border: none;"></a>
 
-The [API Discovery](overview.md) module automatically uncovers rogue (shadow, orphan and zombie) APIs by comparing the discovered API inventory with customers' provided specifications. You upload your API specifications in the [**API Specifications**](#upload-specifications-and-set-comparison-parameters) section and the module automatically highlights shadow, orphan and zombie APIs.
+The [API Discovery](overview.md) module automatically identifies shadow, orphan and zombie APIs by comparing the specification to the live traffic.
```
Comment (AnastasiaTWW) [inline docs/latest/api-discovery/rogue-api.md:3]: the specification - what is "THE speicfication"?

File: docs/latest/api-discovery/rogue-api.md (hunk 8 above)
Comment (AnastasiaTWW) [reply docs/latest/api-discovery/rogue-api.md]: definitely not. I do not host my spec anywhere when I upload it from my computer. as for uploading spec from some URL - I do not know where it is the truth to say about URL where I host my spec

**Pull request merged**